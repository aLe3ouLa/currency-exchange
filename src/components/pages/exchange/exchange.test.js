import React from "react";
import ReactDOM from "react-dom";
import Exchange from "./index";
import { Provider } from "react-redux";
import { cleanup, render } from "@testing-library/react";

import { ThemeProvider } from "styled-components";
import { theme, Defaults } from "../../../styles";
import { BrowserRouter as Router } from "react-router-dom";
import configureStore from "redux-mock-store";
const mockStore = configureStore([]);

describe("Exchange Page tests", () => {
  afterEach(cleanup);

  let store;
  beforeEach(() => {
    store = mockStore({
      rates: {
        rates: {
          base: "GBP",
          rates: {},
          date: new Date(),
        },
      },
      active: {
        active: {
          fromCurrency: "EUR",
          fromValue: 0,
          toCurrency: "GBP",
          toValue: 0,
        },
      },
      balance: {
        balance: {
          EUR: 12.01,
          GBP: 0.0,
          USD: 10.01,
          JPY: 0.0,
          lastExchange: {
            fromCurrency: "GBP",
            fromValue: 0,
            toCurrency: "EUR",
            toValue: 0,
            date: new Date(),
          },
        },
      },
      isLoading: false,
    });
  });

  it("renders the Rate page component without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <Provider store={store}>
        <Router>
          <ThemeProvider theme={theme}>
            <React.Fragment>
              <Defaults />
              <Exchange />
            </React.Fragment>
          </ThemeProvider>
        </Router>
      </Provider>,
      div
    );
  });

  it("has a valid snapshot ", () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Router>
          <ThemeProvider theme={theme}>
            <React.Fragment>
              <Defaults />
              <Exchange />
            </React.Fragment>
          </ThemeProvider>
        </Router>
      </Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
