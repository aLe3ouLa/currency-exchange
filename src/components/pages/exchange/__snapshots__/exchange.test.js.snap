// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exchange Page tests has a valid snapshot  1`] = `
<DocumentFragment>
  <div>
    <div
      class="sc-fznyAO gZCuon"
    >
      <div
        class="sc-AxmLO jZWDuR"
      >
        <div
          class="sc-fzozJi wjHre"
        >
          <select
            class="sc-AxheI kUgSbB"
            data-testid="select"
            id="EUR"
            name="EUR"
          >
            <option
              label="GBP"
              value="GBP"
            />
            <option
              label="EUR"
              value="EUR"
            />
            <option
              label="USD"
              value="USD"
            />
            <option
              label="JPY"
              value="JPY"
            />
          </select>
          <div
            class="sc-fzoLsD hxsArJ"
          >
            <span
              class="sc-AxgMl varXG"
            />
            <input
              aria-label="form-input"
              autocomplete="off"
              class="sc-AxhUy dMEhOH"
              name="value"
              placeholder="0"
              type="number"
              value=""
            />
          </div>
        </div>
        <p
          class="sc-Axmtr dPDYyY"
        >
          Balance: € 12.01
        </p>
      </div>
      <button
        class="sc-fzpans fsIHOC"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-exchange-alt fa-w-16 fa-rotate-90 sc-fzplWN fwruor"
          data-icon="exchange-alt"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 168v-16c0-13.255 10.745-24 24-24h360V80c0-21.367 25.899-32.042 40.971-16.971l80 80c9.372 9.373 9.372 24.569 0 33.941l-80 80C409.956 271.982 384 261.456 384 240v-48H24c-13.255 0-24-10.745-24-24zm488 152H128v-48c0-21.314-25.862-32.08-40.971-16.971l-80 80c-9.372 9.373-9.372 24.569 0 33.941l80 80C102.057 463.997 128 453.437 128 432v-48h360c13.255 0 24-10.745 24-24v-16c0-13.255-10.745-24-24-24z"
            fill="currentColor"
          />
        </svg>
      </button>
      <div
        class="sc-AxirZ gumsER"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-circle-notch fa-w-16 fa-spin "
          data-icon="circle-notch"
          data-prefix="fas"
          data-testid="spinner"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M288 39.056v16.659c0 10.804 7.281 20.159 17.686 23.066C383.204 100.434 440 171.518 440 256c0 101.689-82.295 184-184 184-101.689 0-184-82.295-184-184 0-84.47 56.786-155.564 134.312-177.219C216.719 75.874 224 66.517 224 55.712V39.064c0-15.709-14.834-27.153-30.046-23.234C86.603 43.482 7.394 141.206 8.003 257.332c.72 137.052 111.477 246.956 248.531 246.667C393.255 503.711 504 392.788 504 256c0-115.633-79.14-212.779-186.211-240.236C302.678 11.889 288 23.456 288 39.056z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        class="sc-AxmLO jvamAq"
      >
        <div
          class="sc-fzozJi wjHre"
        >
          <select
            class="sc-AxheI kUgSbB"
            data-testid="select"
            id="GBP"
            name="GBP"
          >
            <option
              label="GBP"
              value="GBP"
            />
            <option
              label="EUR"
              value="EUR"
            />
            <option
              label="USD"
              value="USD"
            />
            <option
              label="JPY"
              value="JPY"
            />
          </select>
          <div
            class="sc-fzoLsD hxsArJ"
          >
            <span
              class="sc-AxgMl varXG"
            />
            <input
              aria-label="form-input"
              autocomplete="off"
              class="sc-AxhUy dMEhOH"
              name="value"
              placeholder="0"
              type="number"
              value=""
            />
          </div>
        </div>
        <p
          class="sc-Axmtr dPDYyY"
        >
          Balance: £ 0
        </p>
      </div>
    </div>
    <button
      class="sc-AxjAm FtxYm"
      disabled=""
    >
      Exchange
    </button>
  </div>
</DocumentFragment>
`;
